### PR TITLE
[improve][dependency] Remove jul-to-slf4j

### PR DESCRIPTION
**Pull request by tisonkun**
**I'm unsure whether I should create a proposal or start a discussion about this removal on dev@.**

### Motivation

see also https://lists.apache.org/thread/65skwo492w2nfjwhb3d9y51roq13h8bs

PulsarAdminImpl has a static block requires classpath contains exact either of:

* slf4j-jdk14
* jul-to-slf4j

It causes an issue that if user depends on neither slf4j-jdk14 nor jul-to-slf4j, PulsarAdminImpl will panic with NoClassDefFoundError. And thus, user should add one of them (basically jul-to-slf4j) even if they don't depend on it effectively.

### Modifications

Remove jul-to-slf4j from all modules under Pulsar.

The Pulsar project uses slf4j as the logging facade consistently. If a user want to add a dependency using a different logging framework, they should take care of the packaging strategy themselves.

pulsar-sql has a dependency to slf4j-jdk14 which redirect slf4j to jul. Since this is a unidirectional redirection, it won't cause runtime error, but pulsar-sql will logging with jul framework, which is the same as with previous workaround.

cc @merlimat as the original author of this stuff.
cc @syhily who points me out this issue.
cc @codelipenghui @eolivelli 

### Verifying this change

- [x] Make sure that the change passes the CI checks.

This change is already covered by existing tests.

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes)

The motivation and affected scope are described above.

### Documentation

- [x] `doc-not-needed`

### CODE REVIEW

File: distribution/server/src/assemble/LICENSE.bin.txt [hunk 1]
```diff
@@ -558,7 +558,6 @@ BSD 2-Clause License
 MIT License
  * Java SemVer -- com.github.zafarkhaja-java-semver-0.9.0.jar -- licenses/LICENSE-SemVer.txt
  * SLF4J -- licenses/LICENSE-SLF4J.txt
-    - org.slf4j-jul-to-slf4j-1.7.32.jar
     - org.slf4j-slf4j-api-1.7.32.jar
     - org.slf4j-jcl-over-slf4j-1.7.32.jar
```
Comment (tisonkun) [inline distribution/server/src/assemble/LICENSE.bin.txt:562]: We need to redirect jcl-over-slf4j due to https://github.com/apache/pulsar/pull/1740 that it's an internal dependency to commons-logging and we should handle within Pulsar itself.

**Comment by tisonkun** [comment]
Pulsar Bot Command match exactly without whitespace, perhaps we should trim before match comments..

**Comment by tisonkun** [comment]
> Pulsar Bot Command match exactly without whitespace, perhaps we should trim before match comments..

FYI https://github.com/apache/pulsar-test-infra/pull/54

**Comment by tisonkun** [comment]
CPP Tests ever passed at https://github.com/apache/pulsar/runs/7192710852?check_suite_focus=true, all later commits are merge commit.

Broker Tests Group 1 ever passed at https://github.com/apache/pulsar/runs/7153335951?check_suite_focus=true, all later commits are merge commit.

**Comment by tisonkun** [comment]
It seems Broker Group 1 suite often timeout.

**Comment by syhily** [comment]
Cool. I have been waiting for this for a loooooong time.

**Comment by tisonkun** [comment]
Merge master tomorrow to retry running tests. Stop continue rerunning failure tests >_<

**Comment by tisonkun** [comment]
OWASP failed on false positive cases, see also https://github.com/eclipse/jetty.project/issues/8161#issuecomment-1178788271. I think it should be resolve on upstream soon later.

**Comment by tisonkun** [comment]
"CI - Unit - Brokers - Broker Group 1" failed over days and the failures seem quick weird; for example, `package-info.java` not found but it's exactly there.

@codelipenghui @eolivelli it seems this patch block by those flaky tests. Could you help to confirm and if it's the case, advice what to make progress?

**Comment by tisonkun** [comment]
@codelipenghui @eolivelli all required CI tasks passed. Could you please help on moving forward this patch?

cc @merlimat

**Pull request merged**